folly/AtomicLinkedList.h

```diff
@@ -63,11 +64,19 @@ class AtomicLinkedList {
    * Note: list must be empty on destruction.
    */
   ~AtomicLinkedList() {
+#ifdef MSVC_NO_NONVOID_ATOMIC_IF
```

Comment:
Maybe reduce this assertion to just:

```
  ~AtomicLinkedList() {
    assert(empty());
  }
```

And avoid duplicating the logic and the ifdefs.